brute.go

```diff
@@ -87,6 +87,24 @@ func (b *bruteRanger) ContainingNetworks(ip net.IP) ([]RangerEntry, error) {
 	return results, nil
 }
 
+// CoveredNetworks returns the list of RangerEntry(s) the given ipnet
+// covers.  That is, the networks that are completely subsumed by the
+// specified network.
+func (b *bruteRanger) CoveredNetworks(network net.IPNet) ([]RangerEntry, error) {
+	entries, err := b.getEntriesByVersion(network.IP)
+	if err != nil {
+		return nil, err
+	}
+	var results []RangerEntry
+	for _, entry := range entries {
+		entrynetwork := entry.Network()
+		if network.Contains(entrynetwork.IP) {
```

Comment:
This is incorrect, e.g. 192.168.0.0/15 is not covered by 192.168.0.0/16, but 192.168.0.0/16 does contain ip 192.168.0.0 (IP member of 192.168.0.0/15), you would need to compare the mask.  I would create a function on the Network struct `ContainsNetwork(Network)` to encapsulate the logic.

Adding the following test case to your test set would show this failure.
```
{
	rnet.IPv4,
	[]string{
		"192.168.0.0/15",
	},
	"192.168.0.0/16",
	[]string{},
	"should fail",
}
```

Reply:
The following might not be optimized but should work
```
func (n Network) ContainsNetwork(n1 Network) bool {
	if len(n.Number) != len(n1.Number) {
		return false
	}
	nMaskSize, _ := n.IPNet.Mask.Size()
	n1MaskSize, _ := n1.IPNet.Mask.Size()
	if n1MaskSize < nMaskSize {
		return false
	}
	return n1.Masked(nMaskSize).Equal(n)
}
```

Reply:
Test case
```
func TestNetworkContainsNetwork(t *testing.T) {
	cases := []struct {
		n1           string
		n2           string
		n1ContainsN2 bool
	}{
		{"192.128.0.0/24", "192.128.0.0/24", true},
		{"192.128.0.0/24", "192.128.0.0/25", true},
		{"192.128.0.0/24", "192.128.0.0/23", false},
		{"192.128.0.0/24", "192.127.0.0/24", false},
		{"192.128.0.0/16", "192.128.199.0/24", true},
	}
	for _, tc := range cases {
		name := fmt.Sprintf("%s contains? %s", tc.n1, tc.n2)
		t.Run(name, func(t *testing.T) {
			_, ipNet1, _ := net.ParseCIDR(tc.n1)
			_, ipNet2, _ := net.ParseCIDR(tc.n2)
			assert.Equal(t, tc.n1ContainsN2, NewNetwork(*ipNet1).ContainsNetwork(NewNetwork(*ipNet2)))
		})
	}
}
```